grpclb/src/test/java/io/grpc/grpclb/GrpclbLoadBalancerTest.java

```diff
@@ -1868,21 +1834,58 @@ public void grpclbBalancerStreamClosedAndRetried() throws Exception {
     inOrder.verify(helper).refreshNameResolution();
 
     // Fast-forward to a moment before the retry, the time spent in the last try is deducted.
-    fakeClock.forwardNanos(10 - 4 - 1);
+    getFastForwardBeforeRetry(10 - 4 - 1, inOrder, expectedInitialRequest);
+
+    // Wrapping up
+    verify(backoffPolicyProvider, times(2)).get();
+    verify(backoffPolicy1, times(2)).nextBackoffNanos();
+    verify(backoffPolicy2, times(1)).nextBackoffNanos();
+    verify(helper, times(4)).refreshNameResolution();
+  }
+
+  private StreamObserver<LoadBalanceResponse> getFastForwardBeforeRetry(int nanos, InOrder inOrder,
+      LoadBalanceRequest expectedInitialRequest) {
+    // Fast-forward to a moment before the retry
+    fakeClock.forwardNanos(nanos);
     verifyNoMoreInteractions(mockLbService);
     // Then time for retry
     fakeClock.forwardNanos(1);
     inOrder.verify(mockLbService).balanceLoad(lbResponseObserverCaptor.capture());
+    StreamObserver<LoadBalanceResponse> lbResponseObserver = lbResponseObserverCaptor.getValue();
     assertEquals(1, lbRequestObservers.size());
-    lbRequestObserver = lbRequestObservers.poll();
+    StreamObserver<LoadBalanceRequest> lbRequestObserver = lbRequestObservers.poll();
     verify(lbRequestObserver).onNext(eq(expectedInitialRequest));
     assertEquals(0, fakeClock.numPendingTasks(LB_RPC_RETRY_TASK_FILTER));
+    return lbResponseObserver;
+  }
 
-    // Wrapping up
-    verify(backoffPolicyProvider, times(2)).get();
-    verify(backoffPolicy1, times(2)).nextBackoffNanos();
-    verify(backoffPolicy2, times(1)).nextBackoffNanos();
-    verify(helper, times(4)).refreshNameResolution();
+  private void getFirstBalancerRpc(LoadBalanceRequest expectedInitialRequest, InOrder inOrder) {
+    // First balancer RPC
+    inOrder.verify(mockLbService).balanceLoad(lbResponseObserverCaptor.capture());
+    StreamObserver<LoadBalanceResponse> lbResponseObserver = lbResponseObserverCaptor.getValue();
+    assertEquals(1, lbRequestObservers.size());
+    StreamObserver<LoadBalanceRequest> lbRequestObserver = lbRequestObservers.poll();
+    verify(lbRequestObserver).onNext(eq(expectedInitialRequest));
+    assertEquals(0, fakeClock.numPendingTasks(LB_RPC_RETRY_TASK_FILTER));
```

Comment:
The assertions and verifications here are very common. Can be done differently?

Reply:
Moved to different method, please review